servers/src/main/java/tachyon/master/file/meta/TTLBucket.java

```diff
@@ -80,4 +82,26 @@ public int compareTo(TTLBucket ttlBucket) {
     }
     return 1;
   }
+
+  /**
+   * Compares to a specific object.
+   *
+   * @param object the object to compare
+   * @return true if object is also {@link TTLBucket} and represents
+   * the same TTLIntervalStartTime
+   */
+  @Override
+  public boolean equals(Object object) {
+    if (object == null || getClass() != object.getClass()) {
+      return false;
+    }
+
+    TTLBucket that = (TTLBucket) object;
+    return Objects.equal(mTTLIntervalStartTimeMs, that.mTTLIntervalStartTimeMs);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(mTTLIntervalStartTimeMs, mFiles);
```

Comment:
@gpang Would you recommend to include ```mFiles``` in the contract? I removed it in the second commit, because it was not part of the ```compareTo``` method.